README.md

```diff
@@ -315,6 +316,7 @@ Each event object has the following attributes:
 * **id** *Required* unique identifier. Example: 23522.
 * **event** *Required* event type. Must be "call" or "return".
 * **thread_id** *Required* identifier of the execution thread. Example: 70340688724000.
+* **timestamp** *Optional* timestamp in seconds elapsed from the start of the recording.
```

Comment:
Is there a required resolution?

I propose that we allow the time stamp to be relative to any “zero” as long as it’s accurate within that. The UI can subtract the “start of AppMap” time if it desires.

Reply:
> Is there a required resolution?

`return.elapsed` doesn't so I didn't specify any here either. It's an optional field anyway :shrug: 

> I propose that we allow the time stamp to be relative to any “zero” as long as it’s accurate within that. The UI can subtract the “start of AppMap” time if it desires.

I considered it but I thought that it would complicate the language needlessly and could be confusing. It's trivial to implement a zero in the agent. It also adds a data point ("how long since the start of the recording until the first event?") which would be unavailable with an arbitrary reference.

Reply:
It's not necessarily super trivial because there may be multiple recordings active at once. It's not common, but it can happen. It seems like the spec could just say "timestamp in seconds" and leave it at that? Any tool that wants to do a layout based on the timestamp will need to apply its own scaling based on the range of data it sees.

Reply:
This sounds reasonable. The example would need to be modified to illustrate that it's not necessarily absolute (epoch) time, which is what some readers could otherwise assume.

(Although I still really like the idea of time from the start of the recording.)